### ISSUE TITLE
A new bot, UpGradle, is in love with refreshVersions

### ISSUE THREAD

**Issue by DanySK** [jmfayard/refreshVersions#178]
Hi,
I recently developed a lightweight bot, named UpGradle, whose goal in life is to scan projects on github, decide if there are updates that can be applied, and open a pull request.

Since I very much appreciate refreshVersions, I added support for it in the bot. It currently deals with two jobs:
- Upgrade the Gradle wrapper
- Upgrade a dependency at a time

You can find the bot [here](https://github.com/DanySK/upgradle/), and an example project using travis to run the bot [here](https://github.com/DanySK/upgradle-bot).

Feedback and contribution is welcome. If you find the bot useful, you might want to propose it as a complementary tool.

(By the way, I tried to join the conversation on slack, but I'm welcomed by a request to "sign in into a workspace": is there a public link?)

**Comment by LouisCAD** [comment]
Hi @DanySK!
Thanks for sharing your work, and glad you enjoy refreshVersions!
Does the bot require Travis CI? If not, a GitHub Actions open source example would be nice :)

You can request an invite on [slack.kotl.in](https://slack.kotl.in), then the link to the channel will work once you're invited.

**Comment by DanySK** [comment]
No, it requires a working java environment and an Internet connection, that's it. I'm shipping it as a plain runnable jar application, running `java -jar appname` should suffice.
I never used gh actions, that's why I provided an example via Travis.

**Comment by miikeat** [comment]
Hi @DanySK 
Nice what you did there! 
I would have loved to use your plugin, but sadly we use bitbucket for our internal Projects :(

**Comment by DanySK** [comment]
@miikeat glad you like it, but darn for the current limitations :disappointed:

Other potential users already requested compatibility with GitLab and Bitbucket for the tool.
In principle, it could be done by abstracting away the API elements UpGradle relies upon and providing different implementations. I unfortunately have very little time to spend on changes that do not quickly benefit my work (I was able to write UpGradle because of the lockdown in Italy, and only because it sped up maintenance in the long run). Nevertheless, if you find time to contribute, I gladly review and merge PRs. @jckoenen (kudos) already contributed with a great piece of work for supporting repository tags.
